.github/workflows/changelog.yml

```
@@ -1,178 +1,194 @@
+
+                        if (as[0] !== bs[0]) {
+                          return as[0] - bs[0];
+                        }
+                        
+                        if (as[1] !== bs[1]) {
+                          return as[1] - bs[1];
+                        }
+                      });
+
+                      const noChangelogNeededLabel = prLabels.data.find(label => label.name === 'no-changelog-needed');
+                      const dependenciesLabel = prLabels.data.find(label => label.name === 'dependencies');
+
+                      // We want to prohibit contributors from directly changing the CHANGELOG.md, it's 
+                      // generated so all changes will be lost during the release process.
+                      // Therefore we only allow the changelog to change together with the version.
+                      // In very rare cases where we generate changes in the changelog without changing the 
+                      // version we will just ignore this failing check.
+                      if (changelogChangesPresent && !versionChangesPresent) {
+                          await createOrUpdateChangelogComment("Please don't edit the CHANGELOG.md manually. We use changie to control the Changelog generation, please use `npx changie new` to create a new changelog entry.");
+                          return;
+                      }
+
+                      if (dependenciesLabel) {
+                          return;
+                      }
+
+                      if (noChangelogNeededLabel) {
+                          if (changesPresent) {
+                              await createOrUpdateChangelogComment("Please remove either the 'no-changelog-needed' label or the changelog entry from this PR.");
+                              return;
+                          }
+                          
+                          // Nothing to complain about, so delete any existing comment
+                          await createOrUpdateChangelogComment("", true);
+                          return;
+                      }
+
+                      // We only want to have a changelog entry for the oldest version this PR will
+                      // land in.
+                      const onlyExpectedChangeVersion = allVersions[0]
+                      const missingChangelogEntry = !changedChangesFiles.some(filePath => filePath.includes("/v"+onlyExpectedChangeVersion+"/"))
+                      const unexpectedChangelogEntry = changedChangesFiles.filter(filePath => !filePath.includes("/v"+onlyExpectedChangeVersion+"/"))
+
+
+                      if (missingChangelogEntry) {
+                          await createOrUpdateChangelogComment(`Please add a changelog entry for in the .changes/v${onlyExpectedChangeVersion} folder. If you believe this change does not need a changelog entry, please add the 'no-changelog-needed' label.`);
+                          return;
+                      }
```

nit: I think these messages are sufficient for internal contributors but it might help external contributors more if the message includes the assumed intent of the user, maybe? That could help them correct the PR/reviewer if they want to target a different release.

E.g. If missingChangelogEntry is true it could be because either the user hasn't included any change files at all, or they have a change file in an old version's folder but are missing a backport label. Currently the message assumes the first scenario.

What do you think about something like this? 

```suggestion
                      if (missingChangelogEntry) {
                          await createOrUpdateChangelogComment(`Currently this PR would target a v${onlyExpectedChangeVersion} release. Please add a changelog entry for in the .changes/v${onlyExpectedChangeVersion} folder, or discuss which release you'd like to target with your reviewer. If you believe this change does not need a changelog entry, please add the 'no-changelog-needed' label.`);
                          return;
                      }
```

---

Here's the example of that scenario: https://github.com/DanielMSchmidt/testing-github-workflows/pull/8#issuecomment-2665421940